### ISSUE TITLE
Feature request: expose `glslang::TShader::Includer` via the new C API

### ISSUE THREAD

**Issue by corporateshark** [KhronosGroup/glslang#2063]
The new C API was added (#2038) which partially implements a mechanism to wrap `glslang::TShader::Includer` into a structure `glsl_include_callbacks_t` containing 3 callback functions. It would be useful to expose access to this wrapper.

The simplest solution is to insert a new field `glsl_include_callbacks_t` into the `glslang_input_t` structure.

**Comment by dneto0** [comment]
If you go down this path, Shaderc has a proven C API for includer support.  See the data and first few calls at https://github.com/google/shaderc/blob/master/libshaderc/include/shaderc/shaderc.h#L318

(But I know @johnkslang disagrees with the partitioning of responsibilities between shaderc_include_type_relative and shaderc_include_time_system)

**Comment by johnkslang** [comment]
I documented the correct operation in comments and provided the correct algorithm, in glslang:

```
        // note "local" vs. "system" is not an "either/or": "local" is an
        // extra thing to do over "system". both might get called, as per
        // the c++ specification.

        // for the "system" or <>-style includes; search the "system" paths.
        virtual includeresult* includesystem(const char* /*headername*/,
                                             const char* /*includername*/,
                                             size_t /*inclusiondepth*/) { return nullptr; }

        // for the "local"-only aspect of a "" include. should not search in the
        // "system" paths, because on returning a failure, the parser will
        // call includesystem() to look in the "system" locations.
        virtual includeresult* includelocal(const char* /*headername*/,
                                            const char* /*includername*/,
                                            size_t /*inclusiondepth*/) { return nullptr; }
```

And, the fundamental algorithm:

```
    bool startWithLocalSearch = // true for "...", false for <...>

    // Find the inclusion, first look in "Local" ("") paths, if requested,
    // otherwise, only search the "System" (<>) paths.
    TShader::Includer::IncludeResult* res = nullptr;
    if (startWithLocalSearch)
        res = includer.includeLocal(filename.c_str(), currentSourceFile.c_str(), includeStack.size() + 1);
    if (res == nullptr || res->headerName.empty()) {
        includer.releaseInclude(res);
        res = includer.includeSystem(filename.c_str(), currentSourceFile.c_str(), includeStack.size() + 1);
    }
```

**Comment by johnkslang** [comment]
That is, after fixing the algorithm a long time ago, the only remaining issue in this space is the names reflect a particular convention, rather than the normative C++ spec.  But, from the above, it should be clear how to use/map the names.

**Comment by arcady-lunarg** [comment]
This was implemented in #3185
